manager/manager.go

```diff
@@ -742,65 +731,30 @@ func (m *Manager) rotateRootCAKEK(ctx context.Context, clusterID string) error {
 		return fmt.Errorf("cluster not found: %s", clusterID)
 	}
 
-	// Try to get the private key from the cluster
-	privKeyPEM := cluster.RootCA.CAKey
-	if len(privKeyPEM) == 0 {
-		// We have no PEM root private key in this cluster.
-		log.G(ctx).Warnf("cluster %s does not have private key material", clusterID)
+	// If we don't have a private key, don't bother decrypting
+	if len(cluster.RootCA.CAKey) == 0 {
 		return nil
 	}
-
-	// Decode the PEM private key
-	keyBlock, _ := pem.Decode(privKeyPEM)
-	if keyBlock == nil {
-		return fmt.Errorf("invalid PEM-encoded private key inside of cluster %s", clusterID)
+	logger := log.G(ctx).WithField("cluster.id", clusterID)
+	decryptedPrivKeyPEM, err := ca.DecryptRootKeyPEM(cluster.RootCA.CAKey)
+	if err != nil {
+		logger.Error(err)
+		return err
 	}
-	// If this key is not encrypted, then we have to encrypt it
-	if !x509.IsEncryptedPEMBlock(keyBlock) {
-		finalKey, err = ca.EncryptECPrivateKey(privKeyPEM, strPassphrase)
-		if err != nil {
-			return err
-		}
-	} else {
-		// This key is already encrypted, let's try to decrypt with the current main passphrase
-		_, err = x509.DecryptPEMBlock(keyBlock, []byte(passphrase))
-		if err == nil {
-			// The main key is the correct KEK, nothing to do here
-			return nil
-		}
-		// This key is already encrypted, but failed with current main passphrase.
-		// Let's try to decrypt with the previous passphrase
-		unencryptedKey, err := x509.DecryptPEMBlock(keyBlock, []byte(passphrasePrev))
-		if err != nil {
-			// We were not able to decrypt either with the main or backup passphrase, error
-			return err
-		}
-		unencryptedKeyBlock := &pem.Block{
-			Type:    keyBlock.Type,
-			Bytes:   unencryptedKey,
-			Headers: keyBlock.Headers,
-		}
 
-		// We were able to decrypt the key, but with the previous passphrase. Let's encrypt
-		// with the new one and store it in raft
-		finalKey, err = ca.EncryptECPrivateKey(pem.EncodeToMemory(unencryptedKeyBlock), strPassphrase)
-		if err != nil {
-			log.G(ctx).Debugf("failed to rotate the key-encrypting-key for the root key material of cluster %s", clusterID)
-			return err
-		}
+	if subtle.ConstantTimeCompare(cluster.RootCA.CAKey, decryptedPrivKeyPEM) != 1 {
+		logger.Warn("Encrypting the root key material via environment variables is deprecated and superceded by default raft log encryption.")
+		// Let's update the key in the cluster object
+		return s.Update(func(tx store.Tx) error {
+			cluster = store.GetCluster(tx, clusterID)
+			if cluster == nil {
+				return fmt.Errorf("cluster not found: %s", clusterID)
+			}
+			cluster.RootCA.CAKey = decryptedPrivKeyPEM
```

Comment:
I don't have a strong opinion here, but could this be a dangerous thing to do if a user is incrementally upgrading their cluster, and this causes the key to be stored in plaintext on the nodes that aren't using an encrypted Raft log yet?

Not that anyone is using the legacy feature, though.